control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -980,49 +888,60 @@ impl CreateVolumeExe for CreateVolume {
     async fn create<'a>(
         &'a self,
         context: &mut Context<'a>,
-        candidates: CreateReplicaCandidate,
+        candidates: Self::Candidates,
     ) -> Vec<Replica> {
-        let mut replicas = Vec::<Replica>::with_capacity(candidates.candidates().len());
-        for replica in candidates.candidates() {
-            if replicas.len() >= self.replicas as usize {
-                break;
-            } else if replicas.iter().any(|r| {
-                r.node == replica.node
-                    || spread_label_is_same(r, replica, context).unwrap_or_else(|error| {
+        let mut fina_replicas = Vec::<Replica>::new();
+        for crc in candidates.iter() {
```

Comment:
Can consume here, rather than reference

```suggestion
        for crc in candidates {
```

Reply:
Also what does crc stand for?

Reply:
create_replica_candiate